src/hotspot/share/memory/metaspace/freeBlocks.cpp

```diff
@@ -47,7 +48,12 @@ MetaWord* FreeBlocks::remove_block(size_t requested_word_size) {
   if (requested_word_size > MaxSmallBlocksWordSize) {
     p = _tree.remove_block(requested_word_size, &real_size);
   } else {
+    // Small blocks: prefer bin lists; if nothing fitting found, take a bigger block
+    // and reuse that.
     p = _small_blocks.remove_block(requested_word_size, &real_size);
+    if (p == nullptr) {
+      p = _tree.remove_block(requested_word_size, &real_size);
+    }
```

Comment:
Reviewer info: this is a functional change, arguably a bug upstream: 
`FreeBlocks` manages small blocks in a linear bin list (`_small_blacks`), larger blocks in a binary tree (`_tree`). Before, when trying to allocate a small block from `FreeBlocks`, we only would look into the bin list. But it makes sense to look into the large block tree too if that fails. Later down, we chop the block in two and save off the remainder anyway.